oauth2client/service_account.py

```diff
@@ -61,31 +61,50 @@ def __init__(self, service_account_id, service_account_email, private_key_id,
   def _generate_assertion(self):
```

Comment:
This shares the code for the the Jwt creation. However, they are also used differently, as JWT Access goes in directly into the authorization headers as a bearer token, while scoped credentials fetch an access token and use that for authorization token. I was expecting to see some logic reflecting that.